### PR TITLE
Fix compiler warning when service string is exactly 8 characters

### PR DESCRIPTION
**Pull request by averne**

### CODE REVIEW

File: include/stratosphere/mitm/mitm_server.hpp [hunk 1]
```diff
@@ -37,7 +37,7 @@ class MitmServer : public IWaitable {
             Handle query_h = 0;
 
             DoWithSmMitmSession([&]() {
-                strncpy(mitm_name, service_name, 8);
+                strncpy(mitm_name, service_name, 9);
```
Comment (mtheall) [inline include/stratosphere/mitm/mitm_server.hpp]: Shouldn't this be `sizeof(mitm_name)`?

**Comment by SciresM** [comment]
Nah, the code is semantically correct at 8.

**Comment by averne** [comment]
Service names are 8 characters max though, right? Add the null terminator and it will be cut by the strncpy, which generates a warning.

**Comment by SciresM** [comment]
@averne The code right now does:

```
                strncpy(mitm_name, service_name, 8);
                mitm_name[8] = '\x00';
```

In the event of an 8-length service name, strncpy will truncate without writing a NULL-terminator. We then add a NULL terminator at 8, and the result is NULL-terminated correctly. There's no actual problem.

**Comment by averne** [comment]
The problem is that the compiler can see the null character being truncated, and will warn against it:
```
warning: 'char* strncpy(char*, const char*, size_t)' output truncated before terminating nul copying 8 bytes from a string of the same length [-Wstringop-truncation]
strncpy(mitm_name, service_name, 8);
```

**Comment by SciresM** [comment]
@averne string truncation is not a problem -- it is a documented behavior of strncpy, and I am intentionally relying on it. Compiler's wrong here, because that behavior is not a problem and I am relying on it.

**Comment by averne** [comment]
I know this code is correct, just trying to squash a warning.

**Comment by SciresM** [comment]
Does this warning impact any current Atmosphere code?

I don't want to change the strncpy size to 9 -- that is not my intent as a programmer. I mean "copy up to 8 bytes, and do not append a NULL terminator if copying exactly 8", which is what strncpy does.

I'd be fine in theory with disabling warnings via pragma, but that seems kind of dumb. I'd rather just have the warning, since the warning is wrong.

**Comment by roblabla** [comment]
Disabling the warning with a pragma around this code is the right thing to do IMO. It's acknowledging that you are relying on this behavior, and that the code is working as intended, there is no bug.

**Comment by averne** [comment]
AMS is currently not affected, my code is.
I've pushed the pragmas, do what you will with that.

**Pull request merged**